### PR TITLE
Add *AnyValue functions

### PR DESCRIPTION
**Pull request by H-ZeX**
## What have you changed? (mandatory)

- Add IntAnyValue, RealAnyValue, StringAnyValue, DurationAnyValue, TimeAnyValue, DecimalAnyValue, JsonAnyValue, and corresponding unit tests.
- According to MySQL(8.0)' `ANY_VALUE` function's [doc](https://dev.mysql.com/doc/refman/8.0/en/miscellaneous-functions.html#function_any-value)
    > The function return value and type are the same as the return value and type of its argument, but the function result is not checked for the ONLY_FULL_GROUP_BY SQL mode
- Just simply pick the first element if `self.children.len()>0`, otherwise return `Ok(None)`.

## What are the type of the changes? (mandatory)

- New feature (change which adds functionality)

## How has this PR been tested? (mandatory)

- unit test

## Does this PR affect documentation (docs) or release note? (mandatory)

- no 

## Does this PR affect tidb-ansible update? (mandatory)

- no

## Refer to a related PR or issue link (optional)

- https://github.com/tikv/tikv/issues/3275

### CODE REVIEW

**Comment by breeswish** [comment]
@AndreMouche PTAL, thanks!

File: src/coprocessor/dag/expr/builtin_miscellaneous.rs [hunk 1]
```diff
@@ -12,6 +14,82 @@ const IPV4_LENGTH: usize = 4;
 const PREFIX_COMPAT: [u8; 12] = [0x00; 12];
 
 impl ScalarFunc {
+    pub fn int_any_value(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<i64>> {
+        if self.children.is_empty() {
```
Comment (kennytm) [inline src/coprocessor/dag/expr/builtin_miscellaneous.rs]: I'd prefer writing these without panic possibility:

```rust
if let Some(child) = self.children.first() {
    child.eval_int(ctx, row)
} else {
    Ok(None)
}
```

or maybe through combinators (but it is harder to read):

```rust
self.children.first().map_or(Ok(None), |child| child.eval_int(ctx, row))
```

**Comment by breeswish** [comment]
Thanks a lot

**Comment by H-ZeX** [comment]
Thank you very mush for your review!

**Comment by breeswish** [comment]
Nice job!

**Pull request merged**